preview/MsixCore/MsixCoreInstallerLib/PrepareDevirtualizedRegistry.cpp

```diff
@@ -0,0 +1,188 @@
+    uintmax_t numRemoved = std::experimental::filesystem::remove_all(packageDirectoryPath, error);
+
+    TraceLoggingWrite(g_MsixTraceLoggingProvider,
+        "Removed directory",
+        TraceLoggingValue(packageDirectoryPath.c_str(), "PackageDirectoryPath"),
+        TraceLoggingValue(error.value(), "Error"),
+        TraceLoggingValue(numRemoved, "NumRemoved"));
+
+    return S_OK;
+}
+
+HRESULT PrepareDevirtualizedRegistry::ConvertVfsNameToFullPath(std::wstring fileName, std::wstring& fileFullPath)
+{
+    //The following code gets remainingFilePath from "VFS\FirstDir\...\file.ext" to "\...\file.ext"
+    std::wstring remainingFilePath = fileName;
+    MsixCoreLib_GetPathChild(remainingFilePath); // remove the VFS directory
+
+    std::map<std::wstring, std::wstring> map = FilePathMappings::GetInstance().GetMap();
+    for (auto& pair : map)
+    {
+        if (remainingFilePath.find(pair.first) != std::wstring::npos)
+        {
+            MsixCoreLib_GetPathChild(remainingFilePath); // remove the FirstDir directory.
+
+            // Pre-pend the VFS target directory to obtain the full path for the target location
+            fileFullPath = pair.second + std::wstring(L"\\") + remainingFilePath;
+
+            //Stop looping through the list
+            return S_OK;
+        }
+    }
+
+    TraceLoggingWrite(g_MsixTraceLoggingProvider,
+        "Could not find VFS mapping",
+        TraceLoggingLevel(WINEVENT_LEVEL_WARNING),
+        TraceLoggingValue(fileName.c_str(), "FileName"));
+
+    return E_NOT_SET;
+}
+
+HRESULT PrepareDevirtualizedRegistry::RemoveVfsFile(std::wstring fileName)
+{
+    TraceLoggingWrite(g_MsixTraceLoggingProvider,
+        "RemoveVfsFile",
+        TraceLoggingValue(fileName.c_str(), "FileName"));
+
+    std::wstring fullPath;
+    if (FAILED(ConvertVfsNameToFullPath(fileName, fullPath)))
+    {
+        return S_OK;
+    }
+
+    if (!DeleteFile(fullPath.c_str()))
+    {
+        TraceLoggingWrite(g_MsixTraceLoggingProvider,
+            "Unable to Delete file",
+            TraceLoggingLevel(WINEVENT_LEVEL_WARNING),
+            TraceLoggingValue(fullPath.c_str(), "FullPath"),
+            TraceLoggingValue(GetLastError(), "error"));
+    }
+
+    MsixCoreLib_GetPathParent(fullPath);
+
+    // instead of checking if the directory is empty, just try to delete it.
+    // if it's not empty it'll fail with expected error code that we can ignore
+    if (!RemoveDirectory(fullPath.c_str()))
+    {
+        DWORD error = GetLastError();
+        if (error != ERROR_DIR_NOT_EMPTY)
+        {
+            TraceLoggingWrite(g_MsixTraceLoggingProvider,
+                "Unable to Delete directory",
+                TraceLoggingLevel(WINEVENT_LEVEL_WARNING),
+                TraceLoggingValue(fullPath.c_str(), "FullPath"),
+                TraceLoggingValue(GetLastError(), "error"));
+        }
+    }
+
+    return S_OK;
+}
+
+HRESULT PrepareDevirtualizedRegistry::RemoveVfsFiles()

```

Comment:
>RemoveVfsFiles [](start = 38, length = 14)

the intent is to split out registry changes from extractor, all file related stuff should stay in extractor.